service/frontend/service.go

```diff
@@ -408,28 +408,14 @@ func namespaceRPS(
 
 func effectiveRPS(frontendResolver membership.ServiceResolver, hostRPS func() int, globalRPS func() int) float64 {
 	if gRPS := globalRPS(); gRPS > 0 && frontendResolver != nil {
-		hosts := float64(numFrontendHosts(frontendResolver))
-		return float64(gRPS) / hosts
+		if numHosts := frontendResolver.MemberCount(); numHosts > 0 {
```

Comment:
Makes sense. I am just not sure how much actual difference will this make. If ring is partitioned, we can still go over the specified global limit.

Reply:
Yeah, I thought about this more, and it's not really useful if there's an arbitrary partition--only useful if there's a bug or ringpop just isn't started at all.